decoder.go

```diff
@@ -20,18 +32,45 @@ func frameLineSize(frame *C.AVFrame) *C.int {
 	return (*C.int)(unsafe.Pointer(&frame.linesize[0]))
 }
 
-// h264Decoder is a wrapper around ffmpeg's H264 decoder.
-type h264Decoder struct {
-	codecCtx    *C.AVCodecContext
-	srcFrame    *C.AVFrame
-	swsCtx      *C.struct_SwsContext
-	dstFrame    *C.AVFrame
-	dstFramePtr []uint8
+// getStreamInfo opens a stream URL and retrieves its information
+func getStreamInfo(url string) string {
```

Comment:
Can we please get some definition of what the expected behavior is when a url supports both H264 & H265? 
Is one always preferred over the other? 
Is it nondeterministic?
Are there any product requirements we have been given regarding how this should work?